From bc446316523abc91a0117c2bdc733fe497c9f465 Mon Sep 17 00:00:00 2001
From: Alexander Duyck <alexander.h.duyck@intel.com>
Date: Wed, 9 Jan 2013 08:50:42 +0000
Subject: [PATCH 752/858] ixgbe: Fix SR-IOV MTU warning

This change corrects the fact that we were using 1522 to test for the
max frame size in ixgbe_change_mtu and 1518 in ixgbe_set_vf_lpe.  The
difference was the addition of VLAN_HLEN which we only need to add in the case
of computing a buffer size, but not a filter size.

Signed-off-by: Alexander Duyck <alexander.h.duyck@intel.com>
Tested-by: Sibai Li <Sibai.li@intel.com>
Tested-by: Stephen Ko <stephen.s.ko@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit c560451c2009b94820460e06d3f7c258f7234948)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe.h      | 2 --
 drivers/net/ethernet/intel/ixgbe/ixgbe_main.c | 2 +-
 2 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe.h b/drivers/net/ethernet/intel/ixgbe/ixgbe.h
index 2a61889..d268c7b 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe.h
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe.h
@@ -92,8 +92,6 @@
  */
 #define IXGBE_RX_HDR_SIZE IXGBE_RXBUFFER_256
 
-#define MAXIMUM_ETHERNET_VLAN_SIZE (ETH_FRAME_LEN + ETH_FCS_LEN + VLAN_HLEN)
-
 /* How many Rx Buffers do we bundle into one write to the hardware ? */
 #define IXGBE_RX_BUFFER_WRITE 16 /* Must be power of 2 */
 
diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
index 7845ca1..a76483f 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_main.c
@@ -4880,7 +4880,7 @@ static int ixgbe_change_mtu(struct net_device *netdev, int new_mtu)
   */
  if ((adapter->flags & IXGBE_FLAG_SRIOV_ENABLED) &&
      (adapter->hw.mac.type == ixgbe_mac_82599EB) &&
-     (max_frame > MAXIMUM_ETHERNET_VLAN_SIZE))
+     (max_frame > (ETH_FRAME_LEN + ETH_FCS_LEN)))
   e_warn(probe, "Setting MTU > 1500 will disable legacy VFs\n");
 
  e_info(probe, "changing MTU from %d to %d\n", netdev->mtu, new_mtu);
-- 
1.8.3.1